go/worker/compute/executor/committee/node.go

```diff
@@ -841,6 +848,8 @@ func (n *Node) Dispatch(batch transaction.RawBatch) error {
 	}
 	spanInsert.Finish()
 
+	n.commonNode.CrossNode.Lock()
+	defer n.commonNode.CrossNode.Unlock()
```

Comment:
In the section below (sorry can't comment on it), we should use `lastHeader` instead of accessing `n.commonNode.CurrentBlock` to make sure things are consistent (`n.commonNode.CurrentBlock` could change while you are not holding the lock).

After this is changed, the lock can possibly be moved just before the `handleInternalBatchLocked` call (group publish should not require holding the lock).

Reply:
yeah good idea :+1: